pkg/git/checkout.go

```diff
@@ -0,0 +1,22 @@
+package git
+
+// Checkout a branch.
+func Checkout(directory string, branch string) error {
```

Comment:
This is out of curiosity as much as anything else, I'm assuming that at the moment, we can't use something like [`go-git`](https://github.com/go-git/go-git) ?

Reply:
`go-git` doesn't support merging, rebasing or notes:

https://github.com/go-git/go-git/blob/master/COMPATIBILITY.md

`libgit2` I could see being annoying to ensure it is installed.

I *think* most of the git commands we need have been written.